neuros/board/board_interface.py

```diff
@@ -0,0 +1,32 @@
+# neuros.board.board_interface.py
+
+from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
+
+
+class BoardInterface:
+    """Basic interface for EEG board interaction."""
+
+    def __init__(self, board_id: int = BoardIds.SYNTHETIC_BOARD):
+        self.board_id = board_id
+        self._board = None
+
+    def start(self):
+        """Initialize and start the board."""
+        if not self._board:
+            params = BrainFlowInputParams()
+            self._board = BoardShim(self.board_id, params)
+            self._board.prepare_session()
+            self._board.start_stream()
+
+    def stop(self):
+        """Stop and release the board."""
+        if self._board and self._board.is_prepared():
+            self._board.stop_stream()
+            self._board.release_session()
+            self._board = None
+
+    def get_data(self):
```

Comment:
I noticed that get_current_board_data takes a num_samples argument, which implies a timeslice, which in turn implies that processing the data will occur in chunks, which in turn opens up an opportunity for undesired pausing when processing the data.

Would it be useful to have a stream interface on a separate thread? is that supported?

Reply:
Looks like BrainFlow does support ring buffer and background acquisition. I could make a streaming interface, or run acquisition on a background thread. I favor the first option.